src/compiler/emitter.ts

```diff
@@ -1639,10 +1639,12 @@ module ts {
             }
 
             if (root) {
+                currentSourceFile = root;
                 emit(root);

```

Comment:
I don't like it. If anything, you should set it here and `emit` should take no parameters and operate on `currentSourceFile`. But I prefer the way it was.